mitiq/zne/scaling/folding.py

```diff
@@ -296,13 +292,32 @@ def fold_global(
     num_to_fold = int(round(fraction_scale * len(operations) / 2))
 
     if num_to_fold > 0:
-        folded += Circuit(
-            [inverse(operations[-num_to_fold:])], [operations[-num_to_fold:]]
-        )
+
+        # create a list of operations needed for partial layers
+        ops_for_partial_layers = operations[-num_to_fold:]
+
+        # moments needed for partial layers
+        moment_list = []
+        for (i, moment) in enumerate(base_circuit[::-1]):
+            for i in range(len(ops_for_partial_layers)):
+                for op in moment:
+                    if ops_for_partial_layers[i] == op:
+                        moment_list.append(len(base_circuit) - 1 - i)
+
+        # create the partial layers
+        circuit_partial = Circuit()
+
+        for i in range(len(ops_for_partial_layers)):
+            for moment in base_circuit[moment_list[0] :]:
+                new_moment = Moment()
+                for op in moment:
+                    if op == ops_for_partial_layers[i]:
+                        new_moment = new_moment.with_operation(op)
+                circuit_partial.append(new_moment)
+
+        folded += inverse(circuit_partial) + circuit_partial
```

Comment:
I would simplify the code as in suggestion.
I see the idea of your approach, but it involves many loops which may be more difficult to read and maintain.

```suggestion
        # Create the inverse of the final partial circuit
        inverse_partial = Circuit()
        num_partial = 0
        for moment in base_circuit[::-1]:
            new_moment = Moment()
            for op in moment:
                new_moment = new_moment.with_operation(inverse(op))
                num_partial += 1
                if num_partial == num_to_fold:
                    break
            inverse_partial.append(new_moment)
            if num_partial == num_to_fold:
                break
        # Append partially folded circuit
        folded += inverse_partial + inverse(inverse_partial)
```

Reply:
Thanks ! yes, I was worried about the many loops I was using but I wanted to use as much of the original code as I could.

Reply:
I have added an [additional check to make sure the correct operation ](https://github.com/unitaryfund/mitiq/pull/1113/files#diff-58c4c4e3cb99c8097c841f4267e46fe9702e78865350b7eb607653a332f77b52R303)is used in the inverse circuit. I don't know if there's away to reverse the order of operations in a moment because we want the last operation to be added first.

Reply:
I see, good catch on the operation ordering! 

You can use the following to iterate over moment operations in reversed order:
```
for op in moment.operations[::-1]:
     new_moment =...
```
In this way you can get rid of `ops_for_partial_layers`.